### PR TITLE
add: file closing, ROS_HOSTNAME reading

### PR DESCRIPTION
**Pull request by eghignone**
Adding csv file closing on shutdown, and reading of ROS_HOSTNAME if available

### CODE REVIEW

File: nodes/monitor.py [hunk 1]
```diff
@@ -80,8 +83,14 @@ def update(self, node_map):
   if save_to_csv:
     csv_writer = CSVWriter(csv_file_name, source_list)
     node_start_time = rospy.get_rostime()
-
-  this_ip = os.environ.get("ROS_IP")
+    rospy.on_shutdown(csv_writer.close)
+  else:
+    rospy.on_shutdown(lambda: rospy.loginfo("[cpu monitor] shutting down"))
```
Comment (alspitz) [inline nodes/monitor.py]: Could you move this as a regular loginfo call at the bottom after the main while loop? That way, the shutting down message gets printed with or without the CSV writer.

**Comment by alspitz** [comment]
Thanks for this PR, agreed about closing the CSV file.

File: nodes/monitor.py (hunk 1 above)
Comment (eghignone) [reply nodes/monitor.py]: ofc makes totally sense! I also removed the ros hostname getting, as that's more of a local thing not really in line with the oringal idea/ros guidelines probably

**Comment by alspitz** [comment]
Thanks, looks good! I'm not sure about `ROS_HOSTNAME`; I haven't used it much but it seems it's meant to override `ROS_IP`, so if you'd like to add it back feel free to submit another PR.

**Pull request merged**